google/cloud/spark_connect/client/proxy.py

```diff
@@ -161,10 +162,13 @@ def forward_connection(conn_number, conn, addr, target_host):
     This method should be run inside of a daemon thread so that it will not
     block program termination.
     """
-    with conn:
-        with connect_tcp_bridge(target_host) as websocket_conn:
-            backend_socket = bridged_socket(websocket_conn)
-            connect_sockets(conn_number, conn, backend_socket)
+    try:
+        with conn:
+            with connect_tcp_bridge(target_host) as websocket_conn:
+                backend_socket = bridged_socket(websocket_conn)
+                connect_sockets(conn_number, conn, backend_socket)
+    except Exception as e:
```

Comment:
This discards the exception, which means the enclosing loop which invokes this method will never terminate.

Instead, if we want to do this we would need to reraise the exception after catching it (although it would be better to put this log somewhere else, such as within the `connect_tcp_bridge` method)